cores/portduino/AsyncUDP.h

```diff
@@ -0,0 +1,119 @@
+// FIXME: this is a really hacky implementation that just has the things required to make the firmware compile and run.
+
+#ifndef ESPASYNCUDP_H
+#define ESPASYNCUDP_H
+
+#include <signal.h>
+#include "IPAddress.h"
+#include "Print.h"
+#include <functional>
+#include <mutex>
+#include <memory>
+#include <thread>
+#include <uv.h>
+
+class AsyncUDP;
+
+class AsyncUDPPacket final
+{
+private:
+    uint8_t *_data;
+    size_t _len;
+
+protected:
+    AsyncUDPPacket(uint8_t* byte, size_t len) {
+        _data = byte;
+        _len = len;
+    };
+
+public:
+    uint8_t * data() {
+        return _data;
+    };
+    size_t length() {
+        return _len;
+    };
+
+    friend AsyncUDP;
+};
+
+class asyncUDPSendTask final {
+    protected:
+        uint8_t *data;
+        size_t len;
+        IPAddress addr;
+        uint16_t port;
+
+    public:
+        asyncUDPSendTask(uint8_t *data, size_t len, IPAddress addr, uint16_t port) {
+            this->data = (uint8_t*)malloc(len);
+            memcpy(this->data, data, len);
+            this->len = len;
+            this->addr = addr;
+            this->port = port;
+        };
+    
+        ~asyncUDPSendTask() {
+            free(data);
+        };
+
+    friend AsyncUDP;
+};
+
+typedef std::function<void(AsyncUDPPacket& packet)> AuPacketHandlerFunction;
+typedef std::function<void(void * arg, AsyncUDPPacket& packet)> AuPacketHandlerFunctionWithArg;
+
+class AsyncUDP final
+{
+private:
+    std::mutex _handlerMutex;
+    AuPacketHandlerFunction _handler;
+
+    std::mutex _sendQueueMutex;
+    // the queue is used because uv_udp_send is not threadsafe and uv_async can merge multiple calls into one callback
+    std::vector<std::unique_ptr<asyncUDPSendTask>> _sendQueue;
+
+    std::thread _ioThread;
+
+    bool _connected;
+    IPAddress _listenIP;
+
+    uv_loop_t _loop;
+    uv_udp_t _socket;
+    uv_async_t _async;
+
+public:
+    AsyncUDP();
+    ~AsyncUDP() {
+        raise(SIGSEGV); // FIXME: implement closing and teardown
```

Comment:
Probably one of the places to not intentionally segfault, if the teardown gets called by shutting down the daemon

Reply:
https://github.com/meshtastic/framework-portduino/blob/e54db5e057f8f89b920e1f1e35a22380ec3ad153/cores/portduino/Utility.cpp#L10

Reply:
I'll implement teardown so it does not crash when trying to reboot after reloading the config.